addons/mrp_subcontracting_purchase/tests/test_mrp_subcontracting_purchase.py

```diff
@@ -1030,3 +1030,69 @@ def test_mrp_subcontract_modify_date(self):
         with Form(mo) as production_form:
             production_form.date_start = original_mo_start_date
         self.assertEqual(mo.date_start, original_mo_start_date)
+
+    def test_changing_qty_producing_does_not_create_extra_move_lines(self):
+        """This test ensures that during component recording in a subcontracted MO, changing
+        qty_producing takes account of existing reserved move lines and their resupplied lot/quantity
+        to fulfill component consumption, without creating extra lines with empty lots.
+        """
+
+        resupply_sub_on_order_route = self.env['stock.route'].search([('name', '=', 'Resupply Subcontractor on Order')])
+        self.comp3.write({
+            'detailed_type': 'product',
+            'tracking': 'lot',
+            'route_ids': [Command.link(resupply_sub_on_order_route.id)],
+        })
+        self.bom_finished2.bom_line_ids.write({
+            'product_qty': 10,
+        })
+        lot01 = self.env['stock.lot'].create({
+            'name': 'lot-01',
+            'product_id': self.comp3.id,
+        })
+        stock_location = self.env.ref('stock.stock_location_stock')
+        self.env['stock.quant']._update_available_quantity(self.comp3, stock_location, 100, lot_id=lot01)
```

Comment:
```suggestion
        self.env['stock.quant']._update_available_quantity(self.comp3, self.warehouse.lot_stock_id, 100, lot_id=lot01)
```

Avoid using the hardcoded `stock.stock_location_stock (the “nighty tag”)`